test/e2e/telemetry_logs_analysis_test.go

```diff
@@ -25,117 +24,128 @@ import (
+		gomegaMaxLength            = format.MaxLength
+		logLevelsRegexp            = "ERROR|error|WARNING|warning|WARN|warn"
 	)
 
-	makeResourcesOtelCollector := func() []client.Object {
+	makeResourcesTracePipeline := func(backendName string) []client.Object {
 		var objs []client.Object
-		objs = append(objs, kitk8s.NewNamespace(otelCollectorNs).K8sObject())
 
-		// backends
-		otelCollectorLogBackend := backend.New(otelCollectorNs, backend.SignalTypeLogs, backend.WithName(otelCollectorLogBackendName))
-		objs = append(objs, otelCollectorLogBackend.K8sObjects()...)
-		otelCollectorLogbackendExportURL = otelCollectorLogBackend.ExportURL(proxyClient)
-		metricBackend := backend.New(otelCollectorNs, backend.SignalTypeMetrics, backend.WithName(metricBackendName))
-		metricbackendExportURL = metricBackend.ExportURL(proxyClient)
-		objs = append(objs, metricBackend.K8sObjects()...)
-		traceBackend := backend.New(otelCollectorNs, backend.SignalTypeTraces, backend.WithName(traceBackendName))
-		tracebackendExportURL = traceBackend.ExportURL(proxyClient)
+		//backend
+		traceBackend := backend.New(namespace, backend.SignalTypeTraces, backend.WithName(backendName))
+		traceBackendURL = traceBackend.ExportURL(proxyClient)
 		objs = append(objs, traceBackend.K8sObjects()...)
 
-		// log pipeline
-		otelCollectorLogPipelineName = fmt.Sprintf("%s-pipeline", otelCollectorLogBackend.Name())
-		otelCollectorLogPipeline := testutils.NewLogPipelineBuilder().
-			WithName(otelCollectorLogPipelineName).
-			WithIncludeNamespaces(kitkyma.SystemNamespaceName).
-			WithIncludeContainers("collector").
-			WithHTTPOutput(testutils.HTTPHost(otelCollectorLogBackend.Host()), testutils.HTTPPort(otelCollectorLogBackend.Port())).
+		//pipeline
+		tracePipeline := testutils.NewTracePipelineBuilder().
+			WithName(backendName).
+			WithOTLPOutput(testutils.OTLPEndpoint(traceBackend.Endpoint())).
 			Build()
-		objs = append(objs, &otelCollectorLogPipeline)
+		objs = append(objs, &tracePipeline)
+
+		//client
+		objs = append(objs, kitk8s.NewPod("telemetrygen-traces", namespace).WithPodSpec(telemetrygen.PodSpec(telemetrygen.SignalTypeTraces)).K8sObject())
+		return objs
+	}
+
+	makeResourcesMetricPipeline := func(backendName string) []client.Object {
+		var objs []client.Object
+
+		//backend
+		metricBackend := backend.New(namespace, backend.SignalTypeMetrics, backend.WithName(backendName))
+		metricBackendURL = metricBackend.ExportURL(proxyClient)
+		objs = append(objs, metricBackend.K8sObjects()...)
 
-		// metrics & traces
-		metricPipelineName = fmt.Sprintf("%s-pipeline", metricBackend.Name())
+		//pipeline
 		metricPipeline := testutils.NewMetricPipelineBuilder().
-			WithName(metricPipelineName).
-			WithPrometheusInput(true, testutils.IncludeNamespaces(otelCollectorNs)).
-			WithRuntimeInput(true, testutils.IncludeNamespaces(otelCollectorNs)).
-			WithIstioInput(true, testutils.IncludeNamespaces(otelCollectorNs)).
+			WithName(backendName).
```

Comment:
It would require another set of constants (5 pieces) for these names, but they actually bring no real value. We could rename the variable here to "scenarioName" if you like. But I see no benefit in prefixing the pipelines with a "pipeline" string.

Reply:
It was just more for uniformity's sake, since this is how we usually write the e2e tests, but I agree with you, your approach is more straight forward, you can leave it like this.